### PR TITLE
fix: retry failed requests on token refresh

### PR DESCRIPTION
**Pull request by Harasz**
### Description

Fix previously failed request to SSI-Hub on token refresh.

### Contributor checklist

- [x] Breaking changes - check for any existing interfaces changes that are not backward compatible, removed method etc.
- [x] Documentation - document your code, add comments for method, remember to check if auto generated docs were updated.
- [x] Tests - add new or updated existed test for changes you made.
- [x] Migration guide - add migration guide for every breaking change.
- [x] Configuration correctness - check that any configuration changes are correct ex. default URLs, chain ids, smart contract verification on [Volta explorer](https://volta-explorer.energyweb.org/) or [EWC explorer](https://explorer.energyweb.org/).

### CODE REVIEW

File: src/modules/cache-client/cache-client.service.ts [hunk 1]
```diff
@@ -63,44 +64,39 @@ export class CacheClient implements ICacheClient {
...
   async authenticate() {
+    let tokens: AuthTokens | undefined = undefined;
+
+    // First try to refresh access token
     try {
       const refreshedTokens = await this.refreshToken();
 
-      if (refreshedTokens) {
-        getLogger().debug('[CACHE CLIENT] Setting authorization tokens');
-        if (!this.isBrowser) {
```
Comment (JGiter) [inline src/modules/cache-client/cache-client.service.ts:71]: Don't you think it will lead to failed authentication in line 73 in Node.js?

Comment (Harasz) [reply src/modules/cache-client/cache-client.service.ts:71]: I've tested it and everything is fine

File: src/modules/cache-client/cache-client.service.ts [hunk 2]
```diff
@@ -132,7 +128,15 @@ export class CacheClient implements ICacheClient {
       getLogger().debug(`[CACHE CLIENT] axios error unauthorized`);
       const retryOriginalRequest = new Promise((resolve) => {
         this.failedRequests.push(() => {
-          resolve(axios(originalRequest));
+          console.log(originalRequest);
```
Comment (hejkerooo) [inline src/modules/cache-client/cache-client.service.ts:131]: remove console.log

**Review by hejkerooo — APPROVED** [review]

File: src/modules/signer/signer.service.ts [hunk 3]
```diff
@@ -445,8 +445,10 @@ export class SignerService {
    *
```
Comment (JGiter) [inline src/modules/signer/signer.service.ts:445]:
```suggestion
   * @param force when true recalculates token even if it is already present
```

File: src/modules/cache-client/cache-client.service.ts (hunk 1 above)
Comment (JGiter) [reply src/modules/cache-client/cache-client.service.ts:71]: Ok. Lets try it

**Review by JGiter — APPROVED** [review]

File: src/modules/cache-client/cache-client.service.ts [hunk 4]
```diff
@@ -63,44 +64,43 @@ export class CacheClient implements ICacheClient {
...
         'Authorization'
-      ] = `Bearer ${token}`;
+      ] = `Bearer ${tokens.token}`;
     }
-    this.refresh_token = refreshToken;
-    this.pubKeyAndIdentityToken = pubKeyAndIdentityToken;
+    this.refresh_token = tokens.refreshToken;
 
+    // Run previously failed requests
+    console.log(
```
Comment (jrhender) [inline src/modules/cache-client/cache-client.service.ts:101]: Instead of `console.log` here, should it be replaced with `getLogger().log`?

**Review by jrhender — COMMENTED** [review]
Seems like good changes to me, thanks @Harasz  We just need to be sure to test login/logout/change-account scenarios very thoroughly on Switchboard board.

**Comment by Harasz** [comment]
@jrhender I've added jira task to add test for auth scenarios. We don't have one currently. And should be `getLogger`. But I will add task to use logger in other modules.

**Pull request merged**